packages/date-picker/src/vaadin-date-picker-overlay-content-mixin.js

```diff
@@ -775,9 +792,10 @@ export const DatePickerOverlayContentMixin = (superClass) =>
           handled = true;
           break;
         case 'Enter':
-          this._selectDate(this.focusedDate);
-          this._close();
-          handled = true;
+          if (this._selectDate(this.focusedDate)) {
```

Comment:
Here is the change to prevent the `Enter` behavior if we are not allowed to select the focused date.

Reply:
Please also add a test case for pressing enter over a disabled date.

Reply:
...and for any other possibly untested code in the PR

Reply:
I have added a test for pressing enter. I can't think of any other tests relevant to these changes. I'll add a not in the discussion on date shapes to bump that up and see if we can get consensus. Thanks for your help on this PR!